### ISSUE TITLE
Still working?

### ISSUE THREAD

**Issue by markhan101** [CallMeTak/SoundCoreReversing#1]
Hi, I have been researching on Bluetooth rev engineering for a while for my buds liberty 4 NC and happened to find your repo. I was wondering if you are still working on this since your last activity on this was about 8 months ago. I'd love to collaborate on this.

**Comment by CallMeTak** [comment]
I'm not actively working on this, but I can maybe help if you need any assistance.

**Comment by markhan101** [comment]
Then shall I just contact you on this issue and keep it opened?

**Comment by CallMeTak** [comment]
Yeah, that's fine.

**Comment by markhan101** [comment]
Hi are you getting the spp packets by sending commands from the app to the buds? I could not find any spp or att packets even after multiple logs so I'm confused.

**Comment by CallMeTak** [comment]
Yeah, I would connect the earbuds to my phone, open the app and switch through some noise cancellation options, then check the log. It's possible that the Liberty 4 NC sends data differently since I've seen GitHub projects that support other SoundCore products not have support for the Liberty 4 NC. If you share your log file, I can try to look at it, though I'm not sure if it will contain any personal information, so that's up to you. 

And if this route ends up not working you could always try to decompile and reverse engineer the mobile app with something like JadX.

**Comment by markhan101** [comment]
Nah it won't have anything personal, it's just rfcomm etc logs. I'll share it with you soon. Tbh I was thinking about decompiling the app. A little more hassle but would help me more info at once.

**Comment by markhan101** [comment]
[btsnoop_hci.log](https://github.com/user-attachments/files/18331959/btsnoop_hci.log)
This is the snoop bt file, do check it out and let me know. In this file I made sure to switch between the three modes anc, normal and transparency and not play any sound although it did capture a notification sound I believe, Do let me know if you find something.

Thx.

**Comment by CallMeTak** [comment]
I found it. Use this display filter in Wireshark `((btrfcomm) && (bluetooth.dst == e8:ee:cc:93:01:52)) && !(bthfp)`
Looks similar to how it works for all the other SoundCore devices. The 17 byte transfers should be for ANC mode selection just like it is for mine. The 10 byte transfers I never bothered to figure out. But yeah the data is there.
<img width="1250" alt="image" src="https://github.com/user-attachments/assets/c82ebc5a-3a89-4324-95f8-9c7cf3650649" />

**Comment by markhan101** [comment]
Damn, I never thought of looking there. The closest I checked was "Sent UIH Channel=0 -> 15 MPX_CTRL Modem Status Command (MSC)" and I was like if this has something to do with the ANC data bytes then where is it, never realizing it could be 3 or 4 packets down. Anyway, this is a big help and I'm sorry if I was a bother in any way. I'm thinking building a small desktop app to go along with this it would really help with controlling these earbuds. 
Thank you again very much for your help. ❤️

**Comment by CallMeTak** [comment]
No problem, and you weren't a bother. Good luck with your app. I'm going to close this issue now but if you need help again you can open another issue or just contact me on Discord at `_tak.`
